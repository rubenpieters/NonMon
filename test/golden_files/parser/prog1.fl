get2 =
{
     let x <- :get ()
  in let p <- :print (in_0 ())
  in let y <- :get ()
  in return (x, y)
};

handlerRunState =
{ \comp.
  handle comp! with
  | return x -> \s. return (x, s)
  | :get p k -> \s. k! s s
  | :put p k -> \s. k! () p
};

main = {handlerRunState! get2 ()};
