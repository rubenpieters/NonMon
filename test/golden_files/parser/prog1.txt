[TLDeclaration "get2" (ValThunk (ComLet ("x",0) (ComOperationApply "get" ValUnit) (ComLet ("p",0) (ComOperationApply "print" (ValInjection Inj0 ValUnit)) (ComLet ("y",0) (ComOperationApply "get" ValUnit) (ComReturn (ValPair (ValVariable ("x",0)) (ValVariable ("y",0)))))))),TLDeclaration "handlerRunState" (ValThunk (ComLambda ("comp",0) (ComHandle (ComForce (ValVariable ("comp",0))) [HanValClause ("x",0) (ComLambda ("s",0) (ComReturn (ValPair (ValVariable ("x",0)) (ValVariable ("s",0))))),HanOpClause "get" ("p",0) ("k",0) (ComLambda ("s",0) (ComLambdaApply (ComLambdaApply (ComForce (ValVariable ("k",0))) (ValVariable ("s",0))) (ValVariable ("s",0)))),HanOpClause "put" ("p",0) ("k",0) (ComLambda ("s",0) (ComLambdaApply (ComLambdaApply (ComForce (ValVariable ("k",0))) ValUnit) (ValVariable ("p",0))))]))),TLDeclaration "main" (ValThunk (ComLambdaApply (ComLambdaApply (ComForce (ValVariable ("handlerRunState",0))) (ValVariable ("get2",0))) ValUnit))]